### PR TITLE
Fixup auth when using determinate-nixd with long-running builds

### PR DESCRIPTION
**Pull request by cole-h**

### CODE REVIEW

File: magic-nix-cache/src/flakehub.rs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 use crate::env::Environment;
 use crate::error::{Error, Result};
+use crate::DETERMINATE_AUTH_TOKEN;
```
Comment (grahamc) [inline magic-nix-cache/src/flakehub.rs:3]:
```suggestion
use crate::DETERMINATE_AUTH_TOKEN_PATH;
```

File: magic-nix-cache/src/flakehub.rs [hunk 2]
```diff
@@ -137,24 +139,53 @@ pub async fn init_cache(
...
+                // NOTE(cole-h): This is a workaround -- at the time of writing, determinate-nixd
+                // handles the GitHub Actions JWT refreshing for us, which means we don't know when
+                // this will happen. At the moment, it does it roughly every 2 minutes (less than
+                // half of the total lifetime of the issued token), so refreshing every 30 seconds
+                // is "fine".
+                let api_clone = api.clone();
+                let token_file = String::from(DETERMINATE_AUTH_TOKEN);
+                let server_config = ServerConfig {
+                    endpoint: flakehub_cache_server.to_string(),
+                    token: Some(attic_client::config::ServerTokenConfig::File {
```
[the remark this answers is not in the document]
Comment (cole-h) [reply magic-nix-cache/src/flakehub.rs:172]: :shrug: Not our API

File: magic-nix-cache/src/flakehub.rs [hunk 3]
```diff
@@ -137,24 +139,53 @@ pub async fn init_cache(
...
+
+                tokio::task::spawn(refresh_github_actions_jwt_worker(
+                    netrc_path_clone,
+                    initial_github_jwt_clone,
+                    flakehub_cache_server_clone,
+                    api_clone,
+                ));
+            }
+            crate::FlakeHubAuthSource::DeterminateNixd => {
+                // NOTE(cole-h): This is a workaround -- at the time of writing, determinate-nixd
```
Comment (colemickens) [inline magic-nix-cache/src/flakehub.rs:163]: nit: does it make sense to move these closer to where you actually define/set the duration? (for both of these pattern)

File: magic-nix-cache/src/flakehub.rs [hunk 4]
```diff
@@ -137,24 +98,39 @@ pub async fn init_cache(
...
+                let initial_inode = initial_meta.ino();
+
+                tokio::task::spawn(refresh_determinate_token_worker(
+                    netrc_file,
+                    initial_inode,
+                    flakehub_api_server_clone,
+                    flakehub_cache_server_clone,
+                    api_clone,
+                ));
+            }
```
Comment (colemickens) [inline magic-nix-cache/src/flakehub.rs:133]: nit: maybe move this out to unconditionally enable for when dnixd is the auth source.

**Review by colemickens — APPROVED** [review]

**Pull request merged**